crates/sui-keys/src/keystore.rs

```diff
@@ -47,16 +48,25 @@ pub trait AccountKeystore: Send + Sync {
         self.keys().iter().map(|k| k.into()).collect()
     }
 
+    fn create_alias_if_not_exists(&self, alias: &Option<String>) -> Result<String, anyhow::Error>;
+
+    fn aliases(&self) -> Vec<&Alias>;
+    fn aliases_names(&self) -> Vec<&String> {
+        self.aliases().into_iter().map(|x| &x.alias).collect()
+    }
+
     fn generate_and_add_new_key(
         &mut self,
         key_scheme: SignatureScheme,
+        alias: Option<String>,
         derivation_path: Option<DerivationPath>,
         word_length: Option<String>,
-    ) -> Result<(SuiAddress, String, SignatureScheme), anyhow::Error> {
+    ) -> Result<(SuiAddress, String, String, SignatureScheme), anyhow::Error> {
         let (address, kp, scheme, phrase) =
             generate_new_key(key_scheme, derivation_path, word_length)?;
-        self.add_key(kp)?;
-        Ok((address, phrase, scheme))
+        let alias = self.create_alias_if_not_exists(&alias)?;
```

Comment:
`add_key` does not return anything, and `generate_new_key` only returns the `address, kp, scheme, and mnemonic`. 
We need to return the `alias` from this function unless we want to do more calls to the API to get the alias for a specific PK (e.g., when calling `sui client new-address` and want to show the alias too). 
I'll review this and try to only have one call to `create_alias_if_not_exists`. Thanks for your diligence here.

Reply:
I've been digging into this. This would be the cleanest as otherwise `add_key` is called in many places and we'd move the responsibility of `create_alias_if_not_exists` to those places. This would be the only place where we call it twice. The other option is to find an alias by address, and use that. There's still some cloning involved, I will go that route and see what you think.

Reply:
Fixed in https://github.com/MystenLabs/sui/pull/13738/commits/71f6c1662c12478d49580e55be05269dfdc9212e